src/admin/helpers/class-alma-wc-admin-helper-date.php

```diff
@@ -19,17 +19,28 @@ class Alma_WC_Admin_Helper_Date {
 	 *
 	 * @param string $start_date The share of checkout enabled date by the merchant formatted as yyyy-mm-dd.
 	 *
+	 * @param string $format The date format.
+	 *
 	 * @return array
+	 * @throws Exception Exception.
 	 */
-	public static function get_dates_in_interval( $start_date ) {
-		$start_timestamp = strtotime( $start_date );
-
+	public static function get_dates_in_interval( $start_date, $format = 'Y-m-d' ) {
+		// Declare an empty array.
 		$dates_in_interval = array();
-		$time              = time();
-		for ( $i = $start_timestamp; $time - $i > 86400; $i = strtotime( '+1 day', $i ) ) {
-			$dates_in_interval[] = gmdate( 'Y-m-d', $i );
+
+		// Variable that store the date interval of period 1 day.
+		$interval = new DateInterval( 'P1D' );
+
+		$real_end = new DateTime();
+		$real_end->modify( '-1 day' );
+		$period = new DatePeriod( new DateTime( $start_date ), $interval, $real_end );
+
+		// Use loop to store date into array.
+		foreach ( $period as $date ) {
+			$dates_in_interval[] = $date->format( $format );
 		}
 
+		// Return the array elements.
 		return $dates_in_interval;
```

Comment:
Add to PHP client.

Reply:
@syjust @Benjamin-Freoua-Alma @joyet-simon is it ok for everyone ?

Reply:
@ClaireAlma It seams ok